### ISSUE TITLE
[Feature Request] Read flush status from reader

### ISSUE THREAD

**Issue by Karrq** [rousan/async-pipe-rs#1]
Hello!

First of all, thank you so much for the library! I'm gonna use it very soon or for now take inspiration from it, at least.

One thing I'd need and then I'd use is right away is for the reader part to be able to tell if the writer has flushed, and also a way to reset this status. Is it possible? Thank you!

**Comment by rousan** [comment]
> the reader part to be able to tell if the writer has flushed,

 If you look into the source code you will see the `PipeWriter`'s `flush` method is a `no-op` function., It does nothing. Why? Because, whenever you're writing something to the writer, the writer will wait until the reader consumes the data as those are piped and there is no buffer is being used. So, there is no flushing, When the reader completes reading a data chunk, you can be sure that the writer end has written something and it flushed immediately.

> also a way to reset this status. Is it possible? 

did you mean resetting the reader and writing status? no need to reset as you can use those as long as they are alive (not dropped).

**Comment by Karrq** [comment]
I need this functionality because my writer flushes when it's done with a certain communication, but the communication itself can be 1 or more calls to `poll_write`. I do not have control over this writer.

by `reset this status` I meant something like the following:
```
flushed = false;
...
writer calls poll_flush, flushed = true
...
reader calls is_flushed? returns true
reader calls is_flushed? returns true (intentional multiple, even tho not needed for my use case)
...
reader calls clear_flushed, flushed = false
```

**Comment by rousan** [comment]
I think I got your point and it makes sense. I will add this feature as soon as possible. The following methods will be added:

PipeWriter::is_flushed() -> bool
PipeReader::is_flushed() -> bool

and `clear_flushed ` is not needed. Whenever the writer writes something the `is_flushed` flag will be reset.

**Comment by rousan** [comment]
@Karrq  Released `v0.1.3` with the patches.
